tests/test_flow.py

```diff
@@ -123,6 +124,25 @@ def test_authorization_url_access_type(self, instance):
                 code_challenge='2yN0TOdl0gkGwFOmtfx3f913tgEaLM2d2S0WlmG1Z6Q',
                 code_challenge_method='S256')
 
+    def test_authorization_url_generated_verifier(self, instance):
+        scope = 'scope_one'
+        instance.oauth2session.scope = [scope]
+        authorization_url_path = mock.patch.object(
+                instance.oauth2session, 'authorization_url',
+                wraps=instance.oauth2session.authorization_url)
+
+        with authorization_url_path as authorization_url_spy:
+            url, _ = instance.authorization_url()
+
+            _, kwargs = authorization_url_spy.call_args_list[0]
```

Comment:
The mock has an `assert_called_once_with` helper for this, e.g.:

```python
    authorization_url_spy..assert_called_once_with(code_challenge_method='S256')
```

Reply:
thanks, I can do this for checking `code_challenge_method` but the other values are randomized and I'm just doing size and character type validations but I think I need to unpack the arguments to do those regex comparisons.

Definitely open to suggestions here. Thanks.